models/measurement.py

```diff
@@ -79,154 +70,87 @@ def get_subject(Self, db_session, subject_id, subject_type):
             raise ValueError(f"Unknown subject type: {subject_type}")
 
     def subject_join(subject_type):
-        from models import (
-            Metabolite,
-            Strain,
-            Bioreplicate,
-        )
+        from models import Metabolite, Strain, Bioreplicate
 
         if subject_type == 'metabolite':
-            name = Metabolite.metabo_name.label("subjectName")
-            join = (Metabolite, Measurement.subjectId == Metabolite.id)
+            Subject = Metabolite
         elif subject_type == 'strain':
-            name = Strain.memberName.label("subjectName")
-            join = (Strain, Measurement.subjectId == Strain.strainId)
+            Subject = Strain
         elif subject_type == 'bioreplicate':
-            name = Bioreplicate.bioreplicateId.label("subjectName")
             Subject = aliased(Bioreplicate)
-            join = (Subject, Measurement.subjectId == Subject.bioreplicateUniqueId)
+
+        name = Subject.name.label("subjectName")
+        join = (Subject, Measurement.subjectId == Subject.id)
 
         return (name, join)
 
     @classmethod
-    def insert_from_bioreplicates_csv(Self, db_session, study, csv_string):
-        measurements = []
-
-        for row, time, bioreplicate_uuid, technique in _iterate_over_measurement_csv(db_session, study, csv_string):
-            if technique.subjectType != 'bioreplicate':
-                continue
+    def insert_from_csv_string(Self, db_session, study, csv_string, subject_type):
```

Comment:
The three separate methods that insert measurement data have been mercifully merged into a single one with a "subject type" parameter.